src/EditDialog.cpp

```diff
@@ -52,13 +53,10 @@ void EditDialog::reset()
     ui->editorImage->clear();
     hexEdit->setData(QByteArray());
     oldData = "";
-
-    // Hide the warning about editing non-text data in text mode
-    ui->labelBinaryWarning->setVisible(false);
+    dataType = Null;
```

Comment:
Yeah.  I added it here because the code just above it clears the data source buffers.

If `reset()` was only called from the class constructor, this wouldn't be needed.  But it's called from other places too:

* https://github.com/sqlitebrowser/sqlitebrowser/blob/ced6b57bd81f489d3a1b5c849207816cd06528da/src/MainWindow.cpp#L480
* https://github.com/sqlitebrowser/sqlitebrowser/blob/ced6b57bd81f489d3a1b5c849207816cd06528da/src/MainWindow.cpp#L535